### ISSUE TITLE
Please add UI scaling

### ISSUE THREAD

**Issue by the-architect-2030** [foss42/apidash#466]
**Tell us about the task you want to perform and are unable to do so because the feature is not available**
I am using fractional scaling in Manjaro Linux KDE. When i open apidash, the UI elements are so small. The app would be much more usable if there is an option to upscale the UI so that all the elements in the app are larger. This option should be in the settings from within the app.

**Describe the solution/feature you'd like us to add**
whole UI scaling

**Comment by ashitaprasad** [comment]
Thank you for raising the issue. This is definitely a useful feature. 
Also, a good issue on which people can contribute.

**Comment by SajalDevX** [comment]
I think of using a global scale factor using the Provider package to allow users to choose scaling percentages like 100%, 125%, and 150% in the settings section, using the transform.scale globally and hive to remember the user setting. This feature will ensure that the entire app's UI scales proportionately, enhancing usability on high-DPI displays and addressing the small UI element issue

**Comment by ashitaprasad** [comment]
@SajalDevX The approach looks nice. You can send across a PR in case you are working on the issue.

**Comment by ashitaprasad** [comment]
@SajalDevX The current PR involves a lot of code changes. Adding this feature will require some major rehaul.
First, by creating an API Dash design system with multiple flavours for 3-4 scaling values. Then refactoring the existing code base to plugin this design system. Followed by elegantly passing scale values to render the right flavour.

**Comment by Vedanshi27vishu** [comment]
Hello, I would like to take up this issue. Please assign it to me at your earliest convenience. I look forward to contributing. Thank you.

**Comment by ashitaprasad** [comment]
@Vedanshi27vishu you can raise a PR if you have an approach. Also, read [why we do not assign issues to anyone?](https://github.com/foss42/apidash/blob/main/CONTRIBUTING.md#why-we-do-not-assign-issues-to-anyone)

**Comment by NancyWei123** [comment]
I created a new ui_design_system prototype class to solve the issue. I'm not sure whether my solution and thought are right. What's more, I'm not sure that the system will adapt font size according to width or user can ctrl + and ctrl - to room in and out.
<img width="1920" height="1021" alt="Image" src="https://github.com/user-attachments/assets/8d9d941a-cfe0-4147-b14f-a86a184f4181" />

**Comment by animator** [comment]
You can send a PR @NancyWei123

**Comment by 7vignesh** [comment]
@ashitaprasad is this still open to work on?

**Comment by NancyWei123** [comment]
Yes, I tried to solve the problem before, but it was hard and my pr was rejected.

**Comment by 7vignesh** [comment]
> Yes, I tried to solve the problem before, but it was hard and my pr was rejected.

why was it rejected?

**Comment by NancyWei123** [comment]
#1058  I think that mainly because my coding skill is not enough.

**Comment by NancyWei123** [comment]
I'm new to dart, so I will start with simple issue first. You can work on this issue!
